contracts/token/erc721/extensions/ERC721AccessControl.sol

```diff
@@ -0,0 +1,34 @@
+//SPDX-License-Identifier: Apache 2.0
+pragma solidity ^0.8.0;
+
+import { AccessControlEnumerable } from "@openzeppelin/contracts/access/AccessControlEnumerable.sol";
+
+abstract contract ERC721AccessControl is AccessControlEnumerable {
```

Comment:
I'm not sure if we want to refer to this as `ERC721AccessControl` as then it implies in implements ERC721 interfaces. Maybe we should rename it to just `ImmutableAccessControl` something else?